test/CMakeLists.txt

```diff
@@ -651,14 +648,14 @@ set(IMPLICITGEMM_MLIR_ARGS_F ${IMPLICITGEMM_ARGS} --verbose --disable-backward-d
 set(IMPLICITGEMM_MLIR_ARGS_B ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-weights)
 set(IMPLICITGEMM_MLIR_ARGS_W ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-data)
 
-add_custom_test(test_conv_igemm_mlir_small HALF_ENABLED MLIR_ENABLED GFX908_DISABLED GFX90A_DISABLED
+add_custom_test(test_conv_igemm_mlir_small HALF_ENABLED SKIP_UNLESS_MLIR GFX908_DISABLED GFX90A_DISABLED
```

Comment:
Why is this necessary? What was the problem in the implementation of `MLIR_ENABLED`?

This seems as if we'd still need to keep the MLIR smoke tests after #1228, is that right?

Reply:
> Why is this necessary? What was the problem in the implementation of MLIR_ENABLED?

Due to a logic error in computation of `is_mlir_check`, currently `is_mlir_check` is always false, unless both MLIR_ENABLED attribute and MIOPEN_TEST_MLIR are set. This leads to that all custom tests are disabled by default.

Reply:
> This seems as if we'd still need to keep the MLIR smoke tests after #1228, is that right?

This is out of scope of this PR, but yes, because MLIR tests are intended to test MLIR solvers. If, due to some reason, we will disable MLIR, then these tests should be skipped. This applies to all MLIR tests, not only smoke ones.

Reply:
Sounds good, I will amend the jenkinsfile once your PR is merged.